shared/reducers/fs.js

```diff
@@ -525,17 +525,35 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
         kbfsDaemonStatus.set('online', action.payload.online)
       )
     case FsGen.overallSyncStatusChanged:
-      return state.update('syncingFoldersProgress', syncingFoldersProgress =>
-        action.payload.progress.equals(syncingFoldersProgress)
-          ? syncingFoldersProgress
-          : action.payload.progress
+      return state.update('overallSyncStatus', overallSyncStatus =>
+        overallSyncStatus
+          .update('syncingFoldersProgress', syncingFoldersProgress =>
+            action.payload.progress.equals(syncingFoldersProgress)
+              ? syncingFoldersProgress
+              : action.payload.progress
+          )
+          .update('diskSpaceStatus', oldDiskSpaceStatus => {
+            // TODO: allow for 'warning' status
+            const newDiskSpaceStatus = action.payload.outOfSpace ? 'error' : 'ok'
+            if (newDiskSpaceStatus !== oldDiskSpaceStatus && oldDiskSpaceStatus !== 'error') {
+              notifyDiskSpaceStatus(newDiskSpaceStatus)
+            }
+            return newDiskSpaceStatus
+          })
+          // Unhide the banner if the state we're coming from isn't WARNING.
+          .set(
+            'diskSpaceBannerHidden',
+            overallSyncStatus.diskSpaceBannerHidden && overallSyncStatus.diskSpaceStatus === 'warning'
+          )
       )
     case FsGen.setDriverStatus:
       return state.update('sfmi', sfmi => sfmi.set('driverStatus', action.payload.driverStatus))
     case FsGen.showSystemFileManagerIntegrationBanner:
       return state.update('sfmi', sfmi => sfmi.set('showingBanner', true))
     case FsGen.hideSystemFileManagerIntegrationBanner:
       return state.update('sfmi', sfmi => sfmi.set('showingBanner', false))
+    case FsGen.hideDiskSpaceBanner:
+      return state.update('overallSyncStatus', status => status.set('diskSpaceBannerHidden', false))
```

Comment:
🧐is this supposed to be `true`? If so seems like the one in the `overallSyncStatusChanged` case above should be reversed too

Reply:
It does seem that way. I swear I tested this ._.

Reply:
The one in the above case is correct though. Hidden should be set only if both of the following hold:
 - hidden was already set
 - we are still in warning state